app/policies/gsa18f/procurement_policy.rb

```diff
@@ -11,5 +11,17 @@ def can_create!
       super && self.gsa_if_restricted!
     end
     alias_method :can_new!, :can_create!
+
+    def can_cancel!
+      not_cancelled! && check((approver? || delegate? || requester?) && !purchaser?, 
```

Comment:
I don't think that will happen in practice, since they are 2 different roles, set globally for the app, not per-request as with NCR.